vm-linuxagent-upgrade/azuredeploy.json

```diff
@@ -0,0 +1,229 @@
+            "metadata": {
+            "description": "The admin user name."
+             }
+        },
+        "adminPassword": {
+            "type": "securestring",
+            "metadata": {
+            "description": "The password for admin user."
+             }
+        },
+        "vmSourceImageName": {
+            "type": "string",
+            "defaultValue": "b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-14_04_2_LTS-amd64-server-20150309-en-us-30GB",
+            "metadata": {
+            "description": "The Linux image name, we will user ubuntu 14.04.2 LTS as the default."
+             }
+        },
+        "location": {
+            "type": "String",
+            "defaultValue" : "West US",
+            "metadata": {
+            "description": "The region that Linux VM you want to create for."
+             }
+        },
+        "vmSize": {
+            "type": "string",
+            "defaultValue": "Standard_A0",
+            "metadata": {
+            "description": "The VM size that Linux VM you want to create with."
+             }
+        },
+        "publicIPAddressName": {
+            "type": "string",
+			"defaultValue": "myPublicIP",
+			"metadata": {
+            "description": "The public IP address."
+             }
+        },
+        "vmName": {
+            "type": "string",
+			"defaultValue": "myVMName",
+            "metadata": {
+            "description": "The VM name."
+             }
+        },
+        "virtualNetworkName":{
+            "type" : "string",
+			"defaultValue": "myVNET",
+            "metadata": {
+            "description": "The virtual network name."
+             }
+        },
+        "nicName":{
+            "type" : "string",
+            "defaultValue":"myNIC",
+            "metadata": {
+            "description": "The nic name."
+             }
+        },
+	"LinuxAgentVersion":{
+            "type" : "string",
+            "defaultValue":"2.0.12",
+            "metadata": {
+            "description": "the version of Azure Linux Agent you want to update to, the default verison is 2.0.12"
+             }
+        }
+
+    },
+    "variables": {
+        "addressPrefix":"10.0.0.0/16",
+        "subnet1Name": "Subnet-1",
+        "subnet2Name": "Subnet-2",
+        "subnet1Prefix" : "10.0.0.0/24",
+        "subnet2Prefix" : "10.0.1.0/24",
+        "vmStorageAccountContainerName": "vhds",
+        "publicIPAddressType" : "Dynamic",
+        "storageAccountType": "Standard_LRS",
+        "sourceImageName" : "[concat('/',subscription().subscriptionId,'/services/images/',parameters('vmSourceImageName'))]",
+        "vnetID":"[resourceId('Microsoft.Network/virtualNetworks',parameters('virtualNetworkName'))]",
+        "subnet1Ref" : "[concat(variables('vnetID'),'/subnets/',variables('subnet1Name'))]"
+    },
+    "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[parameters('newStorageAccountName')]",
+      "apiVersion": "2014-12-01-preview",
```

Comment:
Need to move away from this API version to 2015-05-01-preview and the associated breaking change https://github.com/Azure/azure-quickstart-templates/blob/master/breaking-change.md